src/hermes/commands/harvest/file_exists.py

```diff
@@ -0,0 +1,267 @@
+# SPDX-FileCopyrightText: 2025 Helmholtz-Zentrum Dresden-Rossendorf
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileContributor: David Pape
+
+"""Module for the ``FileExistsHarvestPlugin`` and it's associated models and helpers."""
+
+from collections import defaultdict
+from dataclasses import dataclass
+from functools import cache
+from mimetypes import guess_type
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional, Set
+from typing_extensions import Self
+import subprocess
+
+from pydantic import BaseModel
+
+from hermes.commands.harvest.base import HermesHarvestCommand, HermesHarvestPlugin
+
+
+def guess_file_type(path: Path):
+    """File type detection for non-standardised formats.
+
+    Custom detection for file types not yet supported by Python's ``guess_type``
+    function.
+    """
+    # YAML was only added to ``guess_type`` in Python 3.14 due to the MIME type only
+    # having been decided in 2024.
+    # See: https://www.rfc-editor.org/rfc/rfc9512.html
+    if path.suffix in [".yml", ".yaml"]:
+        return ("application/yaml", None)
+
+    # TOML is not yet part of ``guess_type`` due to the MIME type only having been
+    # accepted in October of 2024.
+    # See: https://www.iana.org/assignments/media-types/application/toml
+    if path.suffix == ".toml":
+        return ("application/toml", None)
+
+    # cff is yaml.
+    # See: https://github.com/citation-file-format/citation-file-format/issues/391
+    if path.name == "CITATION.cff":
+        return ("application/yaml", None)
+
+    # use non-strict mode to cover more file types
+    return guess_type(path, strict=False)
+
+
+@dataclass
+class URL:
+    """Basic model of a ``schema:URL``.
+
+    See also: https://schema.org/URL
+    """
+
+    url: str
+
+    @classmethod
+    def from_path(cls, path: Path) -> Self:
+        return cls(url=path.as_uri())
+
+    def as_codemeta(self) -> dict:
+        return {
+            "@type": "schema:URL",
+            "@value": self.url,
+        }
+
+
+@dataclass
+class TextObject:
+    """Basic model of a ``schema:TextObject``.
+
+    See also: https://schema.org/TextObject
+    """
+
+    content_size: str
+    encoding_format: str
+    url: URL
+
+    @classmethod
+    def from_path(cls, path: Path) -> Self:
+        size = str(path.stat().st_size)  # string!
+        type_, _encoding = guess_file_type(path)
+        url = URL.from_path(path)
+        return cls(content_size=size, encoding_format=type_, url=url)
```

Comment:
```suggestion
    def from_path(cls, path: Path) -> Self:
        size = str(path.stat().st_size)  # string!
        last_modified = datetime.datetime.fromtimestamp(path.stat().st_mtime).isoformat()
        try:
            created = datetime.datetime.fromtimestamp(path.stat().st_birthtime).isoformat()
        except AttributeError:
            created = None
        created = datetime.datetime.fromtimestamp(path.stat().st_ctime).isoformat()
        type_, _encoding = guess_file_type(path)
        url = URL.from_path(path)
        return cls(content_size=size, encoding_format=type_, url=url, created=created, last_modified=last_modified)
```
We could get the information on when the file was created and last modified like so and then add it into the to schema fields "dateCreated" and "dateModified" as DateTime values.

Reply:
I did make the stat call more robust  though in 6d8c28f